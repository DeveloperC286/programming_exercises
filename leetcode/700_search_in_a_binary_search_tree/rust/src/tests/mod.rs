use super::*;
use insta::assert_debug_snapshot;

#[test]
fn test_empty_tree() {
    // Given
    let tree = None;
    let searching_for = 1;

    // When
    let actual = search_bst(tree, searching_for);

    // Then
    assert_debug_snapshot!("test_empty_tree", actual);
}

#[test]
fn test_multiple_nodes_tree_found() {
    // Given
    let tree = Some(Rc::new(RefCell::new(TreeNode {
        val: 4,
        right: Some(Rc::new(RefCell::new(TreeNode {
            val: 7,
            left: None,
            right: None,
        }))),
        left: Some(Rc::new(RefCell::new(TreeNode {
            val: 2,
            left: Some(Rc::new(RefCell::new(TreeNode {
                val: 3,
                left: None,
                right: None,
            }))),
            right: Some(Rc::new(RefCell::new(TreeNode {
                val: 1,
                left: None,
                right: None,
            }))),
        }))),
    })));
    let searching_for = 2;

    // When
    let actual = search_bst(tree, searching_for);

    // Then
    assert_debug_snapshot!("test_multiple_nodes_tree_found", actual);
}

#[test]
fn test_multiple_nodes_tree_not_found() {
    // Given
    let tree = Some(Rc::new(RefCell::new(TreeNode {
        val: 4,
        right: Some(Rc::new(RefCell::new(TreeNode {
            val: 7,
            left: None,
            right: None,
        }))),
        left: Some(Rc::new(RefCell::new(TreeNode {
            val: 2,
            left: Some(Rc::new(RefCell::new(TreeNode {
                val: 3,
                left: None,
                right: None,
            }))),
            right: Some(Rc::new(RefCell::new(TreeNode {
                val: 1,
                left: None,
                right: None,
            }))),
        }))),
    })));
    let searching_for = 5;

    // When
    let actual = search_bst(tree, searching_for);

    // Then
    assert_debug_snapshot!("test_multiple_nodes_tree_not_found", actual);
}
