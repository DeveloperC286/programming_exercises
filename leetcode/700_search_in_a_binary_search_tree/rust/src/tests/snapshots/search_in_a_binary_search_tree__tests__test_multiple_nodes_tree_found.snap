---
source: src/tests/mod.rs
expression: actual

---
Some(
    RefCell {
        value: TreeNode {
            val: 2,
            left: Some(
                RefCell {
                    value: TreeNode {
                        val: 3,
                        left: None,
                        right: None,
                    },
                },
            ),
            right: Some(
                RefCell {
                    value: TreeNode {
                        val: 1,
                        left: None,
                        right: None,
                    },
                },
            ),
        },
    },
)
